src/main/java/de/tum/in/www1/artemis/web/rest/settings/ide/IdeSettingsResource.java

```diff
@@ -0,0 +1,117 @@
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMapping;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMappingId;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.settings.IdeRepository;
+import de.tum.in.www1.artemis.repository.settings.UserIdeMappingRepository;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+
+/**
+ * REST controller for managing Ide Settings (IdeSettings).
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class IdeSettingsResource {
+
+    private static final Logger log = LoggerFactory.getLogger(IdeSettingsResource.class);
+
+    private final UserIdeMappingRepository userIdeMappingRepository;
+
+    private final IdeRepository ideRepository;
+
+    private final UserRepository userRepository;
+
+    public IdeSettingsResource(UserIdeMappingRepository userIdeMappingRepository, IdeRepository ideRepository, UserRepository userRepository) {
+        this.userIdeMappingRepository = userIdeMappingRepository;
+        this.ideRepository = ideRepository;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * GET ide-settings: get the predefined IDEs
+     *
+     * @return the ide preferences of the user as IdeMappingDTO
+     */
+    @GetMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<?> getIdesOfUser() {
+        User user = userRepository.getUser();
+        log.debug("REST request to get IDEs of user {}", user.getLogin());
+
+        var ideMappings = userIdeMappingRepository.findAllByUserId(user.getId());
+        var ideRecords = ideMappings.stream().map(x -> new IdeMappingDTO(x.getProgrammingLanguage(), x.getIde())).toArray();
+        log.debug("Successfully queried IDEs of user {}", user.getLogin());
+
+        return ResponseEntity.ok(ideRecords);
+    }
+
+    /**
+     * PUT ide-settings: set the IDE for a programming Language of the user
+     *
+     * @return returns the changed ide preferences of the user as IdeMappingDTO
+     */
+    @PutMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<?> setIde(@RequestParam ProgrammingLanguage programmingLanguage, @RequestBody IdeDTO ide) {
+        User user = userRepository.getUser();
+        log.debug("REST request to set IDE of user {}", user.getLogin());
+        if (ide == null || ide.deepLink() == null || ide.name() == null) {
+            log.error("Invalid request body for IDE setting of user {}", user.getLogin());
+            return ResponseEntity.badRequest().build();
+        }
+        if (!ide.deepLink().contains("://") || !ide.deepLink().contains("{cloneUrl}")) {
+            log.error("Invalid deep link for IDE {} of user {}", ide.deepLink(), user.getLogin());
+            return ResponseEntity.badRequest().build();
+        }
+
+        // find or create the ide
+        var savedIde = ideRepository.findByDeepLink(ide.deepLink()).orElse(null);
```

Comment:
Why do you use the deeplink here? The name should be easier and less error prone, isn't it?

Reply:
the name is not unique so it should not be used for the query
The Ide table should save all IDEs, later even IDEs with a custom deeplink provided by the user. To save less entities Preferences with the same deeplink should point to the same entity